src/Mvc/test/Mvc.FunctionalTests/RoutingDynamicTest.cs

```diff
@@ -99,5 +102,39 @@ public async Task DynamicPage_CanSelectPage()
             Assert.Equal(HttpStatusCode.OK, response.StatusCode);
             Assert.Equal("Hello from dynamic page: /DynamicPage", content);
         }
+
+        [Fact]
+        public async Task AppWithDynamicRouteAndMapRazorPages_CanRouteToRazorPage()
+        {
+            // Regression test for https://github.com/aspnet/AspNetCore/issues/13996
```

Comment:
I'm really not sure how to author a credible unit test for this, but this isn't too awful.

Reply:
Move test to its own class?

Reply:
It kinda made sense to cluster with other "dynamic" routing tests. But I didn't want to edit the startup behavior for the other ones. Let me know if you feel strongly about this and I can move it.